lib/adapter.js

```diff
@@ -0,0 +1,501 @@
+var objectTrackingCallbackInterval =
+  process.env.STRONGLOOP_OBJECT_TRACKING_INTERVAL || 15000;
+
+// Public functions to be exported
+
+// Metrics API Use function
+function use(callback) {
+
+  // The callback needs to fire every 60 seconds with data from the last 60
+  // seconds
+  setInterval(function() {
+    // Get data from last 60 seconds
+    var eventLoopData = getEventLoopData();
+    var cpuData = getCpuData();
+    var httpData = getHTTPData();
+    var heapData = getHeapData();
+    var usedHeap = getGCHeapData();
+    var messageData = getMessageData();
+
+    // Fire callback with data to match the strong-agent docs
+    callback('loop.count', eventLoopData.count);
+    callback('loop.minimum', eventLoopData.minimum);
+    callback('loop.maximum', eventLoopData.maximum);
+    callback('loop.average', eventLoopData.average);
+
+    callback('cpu.system', cpuData.system);
+    callback('cpu.total', cpuData.total);
+    callback('cpu.user', cpuData.user);
+
+    callback('http.connection.count', httpData.count);
+
+    // Callback with probe data of each probe if there have been events
+    var probeData;
+
+    for (var i in probeList) {
+      if (probeMetrics[probeList[i]]) {
+        probeData = getProbeData(probeList[i]);
+        callback(probeList[i] + '.count', probeData.count);
+        callback(probeList[i] + '.average', probeData.average);
+        callback(probeList[i] + '.minimum', probeData.minimum);
+        callback(probeList[i] + '.maximum', probeData.maximum);
+      }
+    }
+
+    if (usedHeap > 0) {
+      callback('gc.heap.used', usedHeap);
+    }
+
+    callback('heap.total', heapData.total);
+    callback('heap.used', heapData.used);
+
+    if (messageData.sent > 0 || messageData.received > 0) {
+      callback('messages.out.count', messageData.sent);
+      callback('messages.in.count', messageData.received);
+    }
+
+  }, callbackInterval).unref();
+
+  setInterval(function() {
+    if (objectTrackingEnabled) {
+      var histogram = appmetrics.getObjectHistogram();
+      var objectCounts = compareHistograms(lastObjectHistogram, histogram);
+      lastObjectHistogram = histogram;
+      // Get object stats
+      for (var type in objectCounts) {
+        var tuple = objectCounts[type];
+        // Don't output events for no change?
+        // XXX(toby) - What's the correct behaviour here?
+        // XXX(toby) - Should we skip "weird" object types?
+        if (tuple.count !== 0 && tuple.size !== 0) {
+          callback('object.' + type + '.count', tuple.count);
+          callback('object.' + type + '.size', tuple.size);
+        }
+      }
+    }
+  }, objectTrackingCallbackInterval).unref();
+}
+
+function start() {
+  if (!appmetricsStarted) {
+    appmetrics = require('appmetrics');
+    appmetrics.enable('eventloop');
+    amapi = appmetrics.monitor();
+    initialise();
+    appmetricsStarted = true;
+  }
+}
+
+function on() {
+  // XXX(sam) Unimplemented features:
```

Comment:
its not clear to me whether we need full support for all strong-agent features, but if we do, these are missing